stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetActivity.kt

```diff
@@ -98,7 +98,7 @@ internal class PaymentSheetActivity : BaseSheetActivity<PaymentSheetResult>() {
         )
     }
 
-    private val viewStateObserver = { viewState: ViewState.PaymentSheet? ->
+    private val buyButtonStateObserver = { viewState: ViewState.PaymentSheet? ->
```

Comment:
instead of having two observers with some identical code, can we just check the value `PaymentSheetViewModel.checkoutIdentifier` in the original viewStateObserver code? I think this makes it more readable and we don't need to create an additional `LiveData` with `PaymentSheetViewModel#getViewStateObservable`

Reply:
I would rather keep the logic out of the activity.  I have found the simpler they are the less expensive Robolectric tests required.   I also have an approach for keeping the checkoutIdentifier private to the viewModel class, which I think is preferred over exposing this variable.